test/integration/simple_operations_test.lua

```diff
@@ -1007,3 +1007,139 @@ pgroup:add('test_partial_result_bad_input', function(g)
     t.assert_str_contains(err.err, 'Space format doesn\'t contain field named "lastname"')
 end)
 
+pgroup:add('test_opts_not_damaged', function(g)
+    -- insert
+    local insert_opts = {timeout = 1, bucket_id = 655, fields = {'name', 'age'}}
+    local new_insert_opts, err = g.cluster.main_server.net_box:eval([[
+        local crud = require('crud')
+
+        local insert_opts = ...
+
+        local _, err = crud.insert('customers', {22, box.NULL, 'Elizabeth', 24}, insert_opts)
+
+        return insert_opts, err
+    ]], {insert_opts})
+
+    t.assert_equals(err, nil)
+    t.assert_equals(new_insert_opts, insert_opts)
+
+    -- insert_object
+    local insert_opts = {timeout = 1, bucket_id = 477, fields = {'name', 'age'}}
+    local new_insert_opts, err = g.cluster.main_server.net_box:eval([[
```

Comment:
Why do you use `eval()` instead of `call()`?

```lua
    -- insert_object
    local result, err = g.cluster.main_server.net_box:call(
            'crud.insert_object', {
                'customers',
                {id = 1, name = 'Fedor', age = 59},
                insert_opts
            }
    )
```

Reply:
If I would use `call()` I could not catch modification of opts table